jwst/datamodels/container.py

```diff
@@ -106,6 +114,12 @@ def __init__(self, init=None, asn_exptypes=None, **kwargs):
         elif is_association(init):
             self.from_asn(init)
         elif isinstance(init, str):
+            self.asn_exptypes = (CORON_EXP_TYPES +
+                                 IMAGE2_NONSCIENCE_EXP_TYPES +
+                                 IMAGE2_SCIENCE_EXP_TYPES +
+                                 SPEC2_SCIENCE_EXP_TYPES +
+                                 list(ACQ_EXP_TYPES))
+
```

Comment:
Instead of being down here in the logic, defining `self.asn_exptypes` should be up top.  Maybe replace
```
self.asn_exptypes = asn_exptypes
```
with
```
self.asn_exptypes = asn_exptypes
if self.asn_exptypes is None:
    self.asn_exptypes = (CORON_EXP_TYPES + IMAGE2_NONSCIENCE_EXP_TYPES
    + IMAGE2_SCIENCE_EXP_TYPES + SPEC2_SCIENCE_EXP_TYPES
    + list(ACQ_EXP_TYPES))
```
or similar.  I.e. leave the possibility of specifying it on the `open()` call, but if not, then use the defaults.

Also, it looks like ACQ_EXP_TYPES is a tuple while the others are strings.  That looks like a bug.